test/perf/kind-cluster/mutinode_emulator.sh

```diff
@@ -0,0 +1,269 @@
+#!/bin/bash
+
+# Set the number of worker nodes here
+num_workers=10
+namespace_instaslice="instaslice-operator-system"
+
+# Create the kind configuration file
+cat <<EOF > kind-config.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+EOF
+
+# Add the worker nodes dynamically based on num_workers
+for i in $(seq 1 $num_workers); do
+  cat <<EOF >> kind-config.yaml
+  - role: worker
+EOF
+done
+
+echo "kind-config.yaml file generated with $num_workers worker nodes."
+
+# Create the Kind cluster
+kind create cluster --config kind-config.yaml
+
+wait_for_nodes_ready() {
+  echo "Waiting for all nodes to be in 'Ready' state..."
+  while true; do
+    not_ready_nodes=$(kubectl get nodes --no-headers | grep -v ' Ready' | wc -l)
+    if [ "$not_ready_nodes" -eq 0 ]; then
+      echo "All nodes are ready!"
+      break
+    else
+      echo "$not_ready_nodes nodes not ready yet. Checking again in 5 seconds..."
+      sleep 10
+    fi
+  done
+}
+
+# Wait for nodes to be ready before applying fake capacity
+wait_for_nodes_ready
+
+echo "deploying cert manager"
+
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.3/cert-manager.yaml
+
+namespace="cert-manager"
+label_selector="app=webhook"
+
+check__cert_manager_pods_running() {
+  pod_statuses=$(kubectl get pods -l "$label_selector" -n "$namespace" \
+    -o go-template='{{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .status.phase }}{{ "\n" }}{{ end }}{{ end }}')
+
+  not_running_pods=$(echo "$pod_statuses" | grep -v "Running")
+
+  if [ -z "$not_running_pods" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+echo "Waiting for all pods in namespace '$namespace' with label '$label_selector' to be in 'Running' state..."
+while true; do
+  if check__cert_manager_pods_running; then
+    echo "All pods are now running!"
+    break
+  else
+    echo "Some pods are not yet running. Checking again in 5 seconds..."
+    sleep 5
+  fi
+done
```

Comment:
We can wait until the cert-manager-webhook pod to be Ready by
``
kubectl wait --for=condition=ready pod -l app=webhook -n cert-manager
``
Or the above `check__cert_manager_pods_running` function can be made generic to accept `label_selector` and `namespace` args so that it would be helpful to even monitor the `controller-manager` pod and also the `instaslice-operator-webhook-service`

Reply:
Done